### PR TITLE
Patch 1 :-  Extend support for mime types with few well known extentions.

### PR DESCRIPTION
**Pull request by Akayeshmantha**

### CODE REVIEW

File: c/httpserver.c [hunk 1]
```diff
@@ -3678,9 +3678,63 @@ char *getMimeType(char *extension, int *isBinary){
...
+    *isBinary = TRUE;
+    return "application/pdf";
+  } else if(!strcmp(extension,"sh")) {
+    *isBinary = TRUE;
+    return "application/x-sh";
+  } else if(!strcmp(extension,"tar")) {
+    *isBinary = TRUE;
+    return "application/x-tar";
+  } else if(!strcmp(extension,"txt")) {
+    *isBinary = TRUE;
```
Comment (1000TurquoisePogs) [inline c/httpserver.c:3724]: txt is not binary.

File: c/httpserver.c [hunk 2]
```diff
@@ -3678,9 +3678,63 @@ char *getMimeType(char *extension, int *isBinary){
...
+    return "application/json";
+  } else if(!strcmp(extension,"jsonld")) {
+    *isBinary = TRUE;
+    return "application/ld+json";
+  } else if(!strcmp(extension,"pdf")) {
+    *isBinary = TRUE;
+    return "application/pdf";
+  } else if(!strcmp(extension,"sh")) {
+    *isBinary = TRUE;
```
Comment (1000TurquoisePogs) [inline c/httpserver.c:3718]: sh is not binary

File: c/httpserver.c [hunk 3]
```diff
@@ -3678,9 +3678,63 @@ char *getMimeType(char *extension, int *isBinary){
...
+    return "application/gzip";
+  } else if(!strcmp(extension,"mp3")) {
+    *isBinary = TRUE;
+    return "audio/mpeg";
+  } else if(!strcmp(extension,"jar")) {
+    *isBinary = TRUE;
+    return "application/java-archive";
+  } else if(!strcmp(extension,"json")) {
+    *isBinary = TRUE;
```
Comment (1000TurquoisePogs) [inline c/httpserver.c:3709]: json is not binary

File: c/httpserver.c [hunk 4]
```diff
@@ -3678,9 +3678,63 @@ char *getMimeType(char *extension, int *isBinary){
...
+    return "image/bmp";
+  } else if(!strcmp(extension,"csv")) {
+    *isBinary = TRUE;
+    return "text/csv";
+  } else if(!strcmp(extension,"doc")) {
+    *isBinary = TRUE;
+    return "application/msword";
+  } else if(!strcmp(extension,"docx")) {
+    *isBinary = TRUE;
```
Comment (1000TurquoisePogs) [inline c/httpserver.c:3697]: docx is not binary

File: c/httpserver.c [hunk 5]
```diff
@@ -3678,9 +3678,63 @@ char *getMimeType(char *extension, int *isBinary){
...
+    return "application/octet-stream";
+  } else if(!strcmp(extension,"bmp")) {
+    *isBinary = TRUE;
+    return "image/bmp";
+  } else if(!strcmp(extension,"csv")) {
+    *isBinary = TRUE;
+    return "text/csv";
+  } else if(!strcmp(extension,"doc")) {
+    *isBinary = TRUE;
```
Comment (1000TurquoisePogs) [inline c/httpserver.c:3694]: doc is not binary

File: c/httpserver.c [hunk 6]
```diff
@@ -3678,9 +3678,63 @@ char *getMimeType(char *extension, int *isBinary){
...
+    return "video/x-msvideo";
+  } else if(!strcmp(extension,"bin")) {
+    *isBinary = TRUE;
+    return "application/octet-stream";
+  } else if(!strcmp(extension,"bmp")) {
+    *isBinary = TRUE;
+    return "image/bmp";
+  } else if(!strcmp(extension,"csv")) {
+    *isBinary = TRUE;
```
Comment (1000TurquoisePogs) [inline c/httpserver.c:3691]: csv is not binary

**Comment by Akayeshmantha** [comment]
@1000TurquoisePogs pushed the changes.

**Pull request merged**